ollama/_utils.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+
+from typing import Any, Callable, List, Mapping, Optional, Union, get_args, get_origin
+from ollama._types import Tool
+from collections.abc import Sequence, Set
+from typing import Dict, Set as TypeSet
+import sys
+
+# Type compatibility layer
+if sys.version_info >= (3, 10):
+  from types import UnionType
+
+  def is_union(tp: Any) -> bool:
+    return get_origin(tp) in (Union, UnionType)
+else:
+
+  def is_union(tp: Any) -> bool:
+    return get_origin(tp) is Union
```

Comment:
When types similar to `int | str` or `<type> | <type> | ...` are used in newer python that maps to `UnionType` which doesn't exist in Python < 3.10. 

The parser has logic to check for unions but older python versions would fail as that typing structure does not exist (Python 3.14 is merging these types as well - https://github.com/python/cpython/pull/105511)

After lots of testing the best I could come up with to keep everything happy was setup the function given python version. It's not my favorite addition but the user experience will be a drastic improvement.